### DISCUSSION TITLE
How to Instantly Update React Hook Form, Nested Checkbox ?

### DISCUSSION THREAD

**Discussion by muhsindeniz** [react-hook-form/react-hook-form#9484] [Q&A] [answered]
Here, after returning my Nested array with map, I added a checkbox inside. If this checkbox is false, the default input field will not be required, but if the checkbox is true, the default input field will be required. But when I click on the checkbox, the required in the fields does not update. How can I do it?

<Checkbox
{...register(`configuration.${index}.required`)}
 name={`configuration.${index}.required`}
colorScheme="green"
 size="lg"
 />

**Comment by bluebill1049** [comment, marked as answer]
use `setValue` API? https://react-hook-form.com/api/useform/setvalue

**Reply by muhsindeniz** [reply]
Thank you :D
